thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
     missing.forEach(urn => {
-      return this._fetchSlice(urn, requestContext);
+      const metricUrn = toMetricUrn(urn);
+      if (!(metricUrn in metricUrnToOffestAndUrn)) {
+        metricUrnToOffestAndUrn[metricUrn] = [];
+      }
+      metricUrnToOffestAndUrn[metricUrn].push([toAbsoluteUrn(urn, context.compareMode).split(':')[2].toLowerCase(), urn]);
     });
+
+    Object.keys(metricUrnToOffestAndUrn).forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+  _fetchRowSlice(metricUrn, context, metricUrnToOffestAndUrn){
+        const range = context.anomalyRange;
+        const offsets = metricUrnToOffestAndUrn[metricUrn].map(tuple => tuple[0]);
+        const urns = metricUrnToOffestAndUrn[metricUrn].map(tuple => tuple[1]);
+        const timezone = moment.tz.guess();
+        const url = `/rootcause/metric/aggregate/batch?urn=${metricUrn}&start=${range[0]}&end=${range[1]}&offsets=${offsets}&timezone=${timezone}`;
+        return fetch(url)
+        .then(checkStatus)
```

Comment:
indentation, let's tab all the `.then` and `.catch`.

Also fyi, we can now use `await/async` instead of `.then`s. I can walk you through what it'll look like. just ping me!

Reply:
documentation: 
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
- https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

Reply:
Changed to await/async!